### PR TITLE
Autoindex

### PR DESCRIPTION
**Pull request by tuan9999**

### CODE REVIEW

**Comment by pde-bakk** [comment]
I'm browsing to localhost:8080/ with autoindex being enabled and its not doing anything because it's still opening the index file.

Also some notes:
- Why are you iteratoring through ALL of the locations and then setting _autoindex to true (this is also never been set to false again btw) instead of just using the autoindex value in the location when you're calling server->getpage() ?
- There's no need to have a _autoindex variable in the responsehandler class, just call server->getpage() and see if it returns -3, and in that case run the handleAutoIndex method.
handleAutoIndex seems solid but I havent been able to test it.

**Comment by tuan9999** [comment]
> I'm browsing to localhost:8080/ with autoindex being enabled and its not doing anything because it's still opening the index file.
> 
> Also some notes:
> 
> * Why are you iteratoring through ALL of the locations and then setting _autoindex to true (this is also never been set to false again btw) instead of just using the autoindex value in the location when you're calling server->getpage() ?
> * There's no need to have a _autoindex variable in the responsehandler class, just call server->getpage() and see if it returns -3, and in that case run the handleAutoIndex method.
>   handleAutoIndex seems solid but I havent been able to test it.

1. Yes you won't see anything because root is set to hmlfiles which contains an index.html. Open up localhost:8080/css or localhost:8080/Downloads (any subdirectory of root) and you'll see the directory listing (if auto index is set to true).
2. I am iterating through all locations because I want to see if auto index is set to some sub-location. This is because Nginx also allows for directory listing to be turned on only for some subdirectory/location (currently it only handles root location which is enough for the evals but I will implement the true-to-form Nginx method when I have the time).

File: srcs/ResponseHandler.cpp [hunk 1]
```diff
@@ -109,6 +110,7 @@ ResponseHandler& ResponseHandler::operator= (const ResponseHandler &rhs) {
 int ResponseHandler::generatePage(request_s& request) {
 	int			fd = -1;
 	struct stat statstruct = {};
+	std::cout << this->_autoindex << std::endl;
```
Comment (skrtks) [inline srcs/ResponseHandler.cpp:113]: Remove this before merging?

File: srcs/Server.cpp [hunk 2]
```diff
@@ -182,7 +182,7 @@ void	Server::setup(int fd) {
 		if (is_first_char(str, '}')) // End of server block
 			break ;
 		get_key_value(str, key, value);
-//		 std::cout << "key = " << key << ", value = " << value << "$" << std::endl;
```
Comment (skrtks) [inline srcs/Server.cpp:185]: Remove these comments?

**Comment by skrtks** [comment]
I saw that ResponseHandler::handleTRANSFER_ENCODING is commented out, but in Connection::sendReply we're still checking for *if (request.transfer_buffer) {* on L213. That means that *request.transfer_buffer* in uninitialised when used. @tuan9999 do we need to uncomment handleTRANSFER_ENCODING?

**Comment by skrtks** [comment]
+ check for the leak I told you about :)

**Pull request merged**